nano/node/repcrawler.cpp

```diff
@@ -252,9 +256,19 @@ void nano::rep_crawler::cleanup_reps ()
 	{
 		// Check known rep channels
 		nano::lock_guard<std::mutex> lock (probable_reps_mutex);
-		for (auto const & rep : probable_reps.get<tag_last_request> ())
+		auto iterator (probable_reps.get<tag_last_request> ().begin ());
+		while (iterator != probable_reps.get<tag_last_request> ().end ())
 		{
-			channels.push_back (rep.channel);
+			if (iterator->channel->get_tcp_endpoint ().address () != boost::asio::ip::address_v6::any ())
```

Comment:
Would it be easier to keep the loop as it was and check this condition below in the removal loop below?

Reply:
It seems we cannot erase channel with empty endpoint by channel key because channel value can be modified outside of rep crawler. I.e. hash_code of channel returned some value (indexed in multiindex), but then became 0: https://github.com/nanocurrency/nano-node/blob/develop/nano/node/transport/tcp.cpp#L34
Probably we can add some random uin64_t to channels for comparision

Reply:
Ok, something to keep in mind